spec/lib/services/notification-api-service-spec.js

```diff
@@ -152,5 +200,53 @@ describe('Http.Api.Notification', function () {
                 notificationApiService.publishTaskProgress(message)
             ).to.be.rejected;
         });
+
+        it('should post an alert notification successfully(from catalog collection)', function () {
+            this.sandbox.restore();
+            waterline.catalogs.findMostRecent.onCall(0).resolves(bmc);
+            waterline.catalogs.findMostRecent.onCall(1).resolves(deviceInfoCatalog);
+            return notificationApiService.redfishAlertProcessing(req)
+                .then(function (resp) {
+                    expect(resp).to.deep.equal(amqpMessage);
+                });
+        });
+
+        it('should post an alert notification successfully(from obms collection)', function () {
+            this.sandbox.restore();
+            waterline.catalogs.findMostRecent.onCall(0).resolves(undefined);
+            waterline.obms.findOne.onCall(0).resolves({confid : {host: "ip"}});
+            waterline.catalogs.findMostRecent.onCall(1).resolves(bmc);
+            waterline.catalogs.findMostRecent.onCall(2).resolves(deviceInfoCatalog);
+            return notificationApiService.redfishAlertProcessing(req)
+                .then(function (resp) {
+                    expect(resp).to.deep.equal(amqpMessage);
+                });
+        });
+
+        it('should NOT post an alert notification for an unrecognized node', function () {
+            this.sandbox.restore();
+            waterline.catalogs.findMostRecent.onCall(0).resolves(undefined);
+            waterline.obms.findOne.onCall(0).resolves(undefined);
+            return notificationApiService.redfishAlertProcessing(req)
+                .catch(function (e) {
```

Comment:
To expect a promise to be caught, you should explicitly fail the test in `then` block. Otherwise, if promise is actually fulfilled (out-of exception), the mocha still marks it as success. We usually do this is following way:
```javascript
it('should expect an error', function(done) {
  return notificationApiService.redfishAlertProcessing(req)
  .then(function() {
    done(new Error('should never run here!'));
  })
  .catch(function(err) {
    expect(e).to.have.property('message').that.equals('unrecognized node');
    done();
  });
});
```